common/app/routes/challenges/redux/step-challenge-epic.js

```diff
@@ -1,26 +1,59 @@
 import types from './types';
-import { goToStep, submitChallenge } from './actions';
+import { goToStep, submitChallenge, updateUnlockedSteps } from './actions';
 import { challengeSelector } from './selectors';
 import getActionsOfType from '../../../../utils/get-actions-of-type';
 
+function unlockStep(challengeId, index, unlockedSteps) {
+  if (!challengeId || !index) {
+    return null;
+  }
+  let current = unlockedSteps[challengeId];
+  let updated;
+  if (current) {
+    updated = unlockedSteps[challengeId] = current.concat([index]);
```

Comment:
You are mutating the state directly. You should never mutate state. State should be updated through the reducer